### ISSUE TITLE
Cannot get form data

### ISSUE THREAD

**Issue by mac3384** [mozilla-services/react-jsonschema-form#110]
### Prerequisites

- [x] I have read the [documentation](https://github.com/mozilla-services/react-jsonschema-form/blob/master/README.md#readme).

### Description

Forms by default are generated without any names to their inputs. Because of this you cannot get the data through any classic means. 

### Steps to Reproduce

1. Generate the form described in the example
2. Try to get data using jQuery ($('form').serialize or serializeArray). Or try to assign names for the inputs using the schema
3. Blocked/Frozen.

#### Expected behavior

To be able to assign names for the inputs, or to document how to get the form data in the onSubmit property.

#### Actual behavior

Generated a form with inputs having no name and no possibility to retrieve the actual data on submit.

### Version

1.3.2

**Comment by n1k0** [comment]
The intended way to retrieve form data is to subscribe to eaither the `onSubmit` or `onChange` events:

```js
const schema = {type: "string"};
const onSubmit = (({formData}) => console.log("form data are", formData));

<Form schema={schema} onSubmit={onSubmit}/>
```

I admit this is poorly documented atm, and we should probably fix this.

Using jQuery for that purpose isn't supported nor is a future intended use case. We shall consider adding `name` attributes to widgets though.

**Comment by n1k0** [comment]
Probably fixed by #117 and better docs. Feel free to reopen if needed.
